src/Sylius/Bundle/CoreBundle/Checkout/Step/FinalizeStep.php

```diff
@@ -89,5 +89,8 @@ protected function completeOrder(OrderInterface $order)
 
         $this->dispatchCheckoutEvent(SyliusCheckoutEvents::FINALIZE_COMPLETE, $order);
         $this->dispatchCheckoutEvent(SyliusOrderEvents::POST_CREATE, $order);
+
+        $cartProvider = $this->getCartProvider();
+        $cartProvider->abandonCart();
```

Comment:
After finalize step we don't need a cart. We don't want to give possibility to change order items.